db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -883,9 +884,19 @@ Status DBImpl::CompactRange(const CompactRangeOptions& options,
               level == 0) {
             output_level = ColumnFamilyData::kCompactToBaseLevel;
           }
+          // if it's a BottommostLevel compaction and `kForce*` compaction is
+          // set, disallow trivial move
+          if (level == max_overlapped_level &&
```

Comment:
The existing conditionals feel a bit too complicated. I wonder if we can decide whether to allow trivial move outside of them, just above the `RunManualCompaction()`. Something like, disable trivial move if `level == max_output_level && output_level != level && options.bottommost_level_compaction ...`.

Reply:
Feel free to land this as is. I don't think the solution proposed above really helps. Calling `RunManualCompaction()` outside the loop when `final_output_level > max_overlapped_level` would probably help, but we can revisit it later as it's better to have something that works.

Reply:
I agree that the logic here is getting more complicated and worth refactoring. I'll commit this change first to catch the next release. And revisit this later.